zebra/rt_netlink.c

```diff
@@ -434,7 +437,14 @@ static int netlink_route_change_read_unicast(struct sockaddr_nl *snl,
 			if (gate)
 				memcpy(&nh.gate, gate, sz);
 
-			rib_add(afi, SAFI_UNICAST, vrf_id, proto,
+			if (index) {
+				ifp = if_lookup_by_index(index,
+							 VRF_UNKNOWN);
+				if (ifp)
+					nh_vrf_id = ifp->vrf_id;
+			}
+
+			rib_add(afi, SAFI_UNICAST, vrf_id, nh_vrf_id, proto,
 				0, flags, &p, NULL, &nh, table, metric,
```

Comment:
my understanding is that vrf route leak mechanism is not yet available on netns.
so I think that lookup mechanism should be put in place only for vrf with backend vrf lite.

at least what I will do is : if (!(vrf_id has backend netns) && index)

Reply:
I would suggest that you stop route leaking *before* we get to this point and just leave this code alone.  It should not care about the underlying mechanism

Reply:
I have some other code in #1633 where I will have to know the kind of backend.
mainly because you can have the same interface name and interface index name for different vrfs.